src/routes/data/handlers.test.ts

```diff
@@ -178,13 +203,109 @@ describe('Data routes', () => {
       const get = request(app)
         .get('/not-a-real-id')
         .set('Range', 'bytes=1-2,4-5')
-        .expect(416);
+        .expect(206)
+        .buffer()
+        .parse((res: any, callback: any) => {
+          res.setEncoding('binary');
+          res.data = '';
+          res.on('data', function (chunk: any) {
+            res.data += chunk;
+          });
+          res.on('end', function () {
+            callback(null, Buffer.from(res.data));
+          });
+        })
+        .then((res: any) => {
+          const boundary = res.boundary;
+          assert.equal(
+            res.get('Content-Type'),
+            `multipart/byteranges; boundary=${boundary}`,
+          );
+
+          // binary response data is in res.body as a buffer
+          assert.ok(Buffer.isBuffer(res.body));
+
+          // As  the server respond with a \r\n linebreak and JS template literals use \n linebreaks, explicit \r\n is needed in the expected response
```

Comment:
Thanks for the explanation here

Reply:
A thought - maybe a multiline string followed by `.split('\n').join('\r\n')` would be more readable?

Reply:
Good suggestion. I believe it will work fine and will make it way more readable :)

Reply:
I used `replace` instead
[`51cb78b` (#161)](https://github.com/ar-io/ar-io-node/pull/161/commits/51cb78bf59e439b97422183521d7acf32872dd96)